sklearn/metrics/classification.py

```diff
@@ -1051,80 +1259,14 @@ def precision_recall_fscore_support(y_true, y_pred, beta=1.0, labels=None,
                       "labels=[pos_label] to specify a single positive class."
                       % (pos_label, average), UserWarning)
 
-    if labels is None:
-        labels = present_labels
-        n_labels = None
-    else:
-        n_labels = len(labels)
-        labels = np.hstack([labels, np.setdiff1d(present_labels, labels,
-                                                 assume_unique=True)])
-
     # Calculate tp_sum, pred_sum, true_sum ###
-
-    if y_type.startswith('multilabel'):
-        sum_axis = 1 if average == 'samples' else 0
-
-        # All labels are index integers for multilabel.
-        # Select labels:
-        if not np.all(labels == present_labels):
-            if np.max(labels) > np.max(present_labels):
-                raise ValueError('All labels must be in [0, n labels). '
-                                 'Got %d > %d' %
-                                 (np.max(labels), np.max(present_labels)))
-            if np.min(labels) < 0:
-                raise ValueError('All labels must be in [0, n labels). '
-                                 'Got %d < 0' % np.min(labels))
-
-        if n_labels is not None:
-            y_true = y_true[:, labels[:n_labels]]
-            y_pred = y_pred[:, labels[:n_labels]]
-
-        # calculate weighted counts
-        true_and_pred = y_true.multiply(y_pred)
-        tp_sum = count_nonzero(true_and_pred, axis=sum_axis,
-                               sample_weight=sample_weight)
-        pred_sum = count_nonzero(y_pred, axis=sum_axis,
-                                 sample_weight=sample_weight)
-        true_sum = count_nonzero(y_true, axis=sum_axis,
-                                 sample_weight=sample_weight)
-
-    elif average == 'samples':
```

Comment:
So if users pass average = 'samples' to binary/multiclass ``precision_recall_fscore_support``, the error message will be ``Samplewise confusion is not useful outside of multilabel classification.``? Seems bit unfriendly. I guess we can raise an error here and keep the original error message.

Reply:
> "Samplewise metrics" might be better.

If you don't want to raise an error in ``precision_recall_fscore_support``, then this seems to be an acceptable solution. I guess few users will notice that we change the error message here.